### PR TITLE
Feature/add prefech hs

### PR DESCRIPTION
**Pull request by JiabinYang**

### CODE REVIEW

File: python/paddle/fluid/transpiler/distribute_transpiler.py [hunk 1]
```diff
@@ -242,11 +242,10 @@ def _transpile_nccl2(self,
 
     def _get_all_remote_sparse_update_op(self, main_program):
         sparse_update_ops = []
-        sparse_update_op_types = ["lookup_table"]
+        sparse_update_op_types = ["lookup_table", "nce"]
```
Comment (seiriosPlus) [inline python/paddle/fluid/transpiler/distribute_transpiler.py]: need to add hs here.

File: paddle/fluid/operators/hierarchical_sigmoid_op.cc [hunk 2]
```diff
@@ -189,23 +216,17 @@ class HierarchicalSigmoidGradOpGradVarTypeInference
...
           ->SetType(framework::proto::VarType::LOD_TENSOR);
-      if (hasBias) {
-        VLOG(30) << "hierarchical_sigmoid_grad op "
-                 << framework::GradVarName("Bias") << " is set to LoDTensor";
-        block->Var(bias_grad_var_name)
-            ->SetType(framework::proto::VarType::LOD_TENSOR);
-      }
+    }
+    if (hasBias) {
+      VLOG(30) << "hierarchical_sigmoid_grad op "
```
Comment (jacquesqiao) [inline paddle/fluid/operators/hierarchical_sigmoid_op.cc:226]: change log level to 3

File: paddle/fluid/operators/distributed/parameter_prefetch.cc [hunk 3]
```diff
@@ -117,6 +117,12 @@ static void MergeMultipleVarsIntoOneBySection(
   auto& id_tensor = scope->FindVar(id_name)->Get<framework::LoDTensor>();
   auto* out_tensor =
       scope->FindVar(out_name)->GetMutable<framework::LoDTensor>();
+
+  PADDLE_ENFORCE_GT(
+      out_tensor->numel(), 0,
```
Comment (jacquesqiao) [inline paddle/fluid/operators/distributed/parameter_prefetch.cc:122]: numel maybe slow, use is_inited instead

**Review by jacquesqiao — APPROVED** [review]
LGTM!

**Pull request merged**